waverecovery-gui.py

```diff
@@ -0,0 +1,274 @@
+import sys
+from argparse import ArgumentParser, Namespace
+from PyQt5.QtWidgets import (
+    QApplication,
+    QMainWindow,
+    QWidget,
+    QVBoxLayout,
+    QLabel,
+    QPushButton,
+    QTextEdit,
+    QFileDialog,
+    QFormLayout,
+    QComboBox,
+    QCheckBox,
+    QSpinBox,
+    QMessageBox,
+    QProgressBar,
+    QHBoxLayout,
+)
+from PyQt5.QtCore import Qt, QThread, pyqtSignal
+import io
+from contextlib import redirect_stdout
+from waveheaderprocessor import WaveHeaderProcessor
+from PyQt5.QtGui import QTextCursor, QTextCharFormat, QColor
+
+class WorkerThread(QThread):
+    update_progress = pyqtSignal(int)
+    update_log = pyqtSignal(str, bool)
+
+    def __init__(self, args):
+        super().__init__()
+        self.args = args
+
+    def run(self):
+        try:
+            with io.StringIO() as buf, redirect_stdout(buf):
+                processor = WaveHeaderProcessor()
+                total_files = processor.repair_audio_file_headers(
+                    self.args.source_path,
+                    self.args.destination_path,
+                    self.args.sample_rate,
+                    self.args.bits_per_sample,
+                    self.args.channels,
+                    self.args.verbose,
+                    self.args.force,
+                    self.args.application,
+                    self.args.offset,
+                    self.args.end_offset,
+                )
+                output = buf.getvalue()
+
+            self.update_progress.emit(total_files)
+            self.update_log.emit(output, False)
+
+        except Exception as e:
+            error_message = f'Error: {str(e)}'
+            self.update_log.emit(error_message, True)
+            QMessageBox.critical(self.args.gui, 'Error', error_message)
+        finally:
+            self.args.gui.restore_button.setDisabled(False)
+
+class WaveRecoveryToolGUI(QMainWindow):
```

Comment:
`waverecovery-gui.py` and `waverecovery-gui_modern.py` are very similar except for the UI styling. Should we keep only the modern UI in the end or do you want to keep both? I would prefer to keep only one because otherwise we end up with lots of duplicated code that is hard to maintain.